### PR TITLE
fix: Ticket Flow Refactoring

### LINKED ISSUE #5608: Ticket PDF: Security Issue, can be accessed by anyone
The ticket PDF for events can be accessed by anyone who can find out the link, e.g. see https://api.eventyay.com/static/media/attendees/tickets/pdf/6ae42516-0a03-4436-b2a6-8a60f6a2b16d/QkJ0VHpPUX/ae067598-2896-4579-b4d9-8d5fa8a34e01.pdf

Expected should only be possible to open it from the actual account.

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #5702 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Ticket pdf link should only be accessible by ticket holder.

#### Changes proposed in this pull request:

- Created a decorator wrapper function to check for unauthorized access 
- Implemented a new route to show PDF of ticket

### CODE REVIEW

File: app/api/auth.py [hunk 1]
```diff
@@ -269,3 +273,20 @@ def change_password():
         "name": user.fullname if user.fullname else None,
         "password-changed": True
     })
+
+
+def authorized_access_ticket(ticket_function):
+    @wraps(ticket_function)
+    def wrapper(*args, **kwargs):
+        if not current_identity.is_anonymous and current_user.id == Order.query.filter_by(identifier=kwargs['identifier']).first().user.id:
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/api/auth.py]: What if ORM throws exception on first?

[the remark this answers is not in the document]
Comment (mrsaicharan1) [reply app/api/auth.py]: Didn't take care of that. Will keep it in a try catch block.

[the remark this answers is not in the document]
Comment (mrsaicharan1) [reply app/api/auth.py]: A check needs to be present if the order is present. Will implement it and revert back.

File: app/api/auth.py [hunk 2]
```diff
@@ -269,3 +273,24 @@ def change_password():
         "name": user.fullname if user.fullname else None,
         "password-changed": True
     })
+
+
+def authorized_access_ticket(ticket_function):
+    @wraps(ticket_function)
+    def wrapper(*args, **kwargs):
+        try:
+            user_id = Order.query.filter_by(identifier=kwargs['identifier']).first().user.id
```
Comment (iamareebjamal) [inline app/api/auth.py]: Don't just catch a generic exception. Just check if it returns none and send 404.

**Comment by mrsaicharan1** [comment]
Yeah, will do that.

**Comment by mrsaicharan1** [comment]
@iamareebjamal Checked if the Order associated returned None. Let me know if anything else is there.

File: app/api/auth.py [hunk 3]
```diff
@@ -113,7 +117,7 @@ def login_user(provider):
                 user.facebook_login_hash = random.getrandbits(128)
                 save_to_db(user)
             return make_response(
-                jsonify(user_id=user.id, email=user.email, oauth_hash=user.facebook_login_hash), 200)
+                jsonify(user_id=user.id, email=user.email, facebook_login_hash=user.facebook_login_hash), 200)
```
Comment (iamareebjamal) [inline app/api/auth.py]: These changes are reverts of the fb OAuth PR

**Comment by mrsaicharan1** [comment]
Unrelated changes have surfaced . Will check it out.

**Comment by iamareebjamal** [comment]
LGTM, have you tested all 3 cases, like Not Found, Unauthorized and Found?

**Comment by mrsaicharan1** [comment]
the `identifier` in `/static/media/attendees/tickets/pdf/<string:identifier>/<string:identifier_hash>/<string:filename>` does not seem to match any of the orders' identifiers. @iamareebjamal can you please check this? It's always returning None.

**Comment by mrsaicharan1** [comment]
@iamareebjamal @CosmicCoder96 Can I get an update on this regarding the issue with `current_user` showing up as `None` always?

**Comment by iamareebjamal** [comment]
Should this be closed now?

**Comment by mrsaicharan1** [comment]
@iamareebjamal Yes we can close this.

**Pull request closed without merging**